gcloud/datastore/connection.py

```diff
@@ -68,13 +70,16 @@ def _request(self, dataset_id, method, data):
             uri=self.build_api_url(dataset_id=dataset_id, method=method),
             method='POST', headers=headers, body=data)
 
-        if headers['status'] != '200':
-            raise Exception('Request failed. Error was: %s' % content)
+        status = headers['status']
+        if status != '200':
+            message = ('Request failed with status code %s. '
+                       'Error was: %s' % (status, content))
+            raise six.moves.http_client.HTTPException(message)
```

Comment:
1) I should have held off on updating this for another commit, my bad.

2) I don't think the burden is worth it unless there is a fast way to map from the standard lib.